rsocket/internal/ConnectionSet.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include "rsocket/internal/ConnectionSet.h"
+
+#include "rsocket/statemachine/RSocketStateMachine.h"
+
+#include <folly/io/async/EventBase.h>
+
+namespace rsocket {
+
+ConnectionSet::ConnectionSet() {}
+
+ConnectionSet::~ConnectionSet() {
+  VLOG(1) << "Started ~ConnectionSet";
+  SCOPE_EXIT { VLOG(1) << "Finished ~ConnectionSet"; };
+
+  StateMachineMap map;
+
+  // Set up the baton and counter of connections yet to be closed.  Move all the
+  // connections out of the synchronized map so we don't block while keeping the
+  // map locked.
+  {
+    auto locked = machines_.lock();
+    if (locked->empty()) {
+      VLOG(2) << "No connections to close, early exit";
+      return;
+    }
+
+    VLOG(2) << "Need to close " << locked->size() << " connections";
+
+    map.swap(*locked);
```

Comment:
Now that we are not waiting for all the stateMachines to close, can we get rid of the copy.  The code can become cleaner (and the destruction will also be faster)

Reply:
Technically this is a move and not a copy.  This way should be faster, as other threads running `RSocketStateMachine::close()` can interleave with `~ConnectionSet`, as the `~ConnectionSet` thread isn't holding the lock.